### ISSUE TITLE
x/vgo: fails trying to open ziphash from cache dir

### ISSUE THREAD

**Issue by artyom** [golang/go#26127]
### What version of Go are you using (`go version`)?

    go version go1.10.3 darwin/amd64 vgo:2018-02-20.1

vgo at revision bbcfaa0c1b10a3e580b80b1d575ce55054b80a23

### What operating system and processor architecture are you using (`go env`)?

`vgo env` output:

```
GOARCH="amd64"
GOBIN=""
GOCACHE="/Users/artyom/Library/Caches/go-build"
GOEXE=""
GOHOSTARCH="amd64"
GOHOSTOS="darwin"
GOOS="darwin"
GOPATH="/Users/artyom/go"
GOPROXY=""
GORACE=""
GOROOT="/Users/artyom/Library/go"
GOTMPDIR=""
GOTOOLDIR="/Users/artyom/Library/go/pkg/tool/darwin_amd64"
GCCGO="gccgo"
CC="clang"
CXX="clang++"
CGO_ENABLED="1"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/var/folders/lb/3rk8rqs53czgb4v35w_342xc0000gn/T/go-build057133119=/tmp/go-build -gno-record-gcc-switches -fno-common"
VGOMODROOT="/Users/artyom/go/src/github.com/artyom/vgo-test2"
```

### What did you do?

Trying to build any package with dependencies, i.e. https://github.com/artyom/vgo-test2

### What did you expect to see?

Build succeeds.

### What did you see instead?

vgo fails trying to open non-existent .ziphash file from cache directory:

```
vgo-test2 (master) ¶ vgo build
vgo: verifying golang.org/x/sync@v0.0.0-20180314180146-1d60e4601c6f: open /Users/artyom/go/src/mod/cache/download/golang.org/x/sync/@v/v0.0.0-20180314180146-1d60e4601c6f.ziphash: no such file or directory

1 vgo-test2 (master) ¶ ls ~/go/src/mod/cache/download/golang.org/x/sync/@v/v0.0.0-20180314180146-1d60e4601c6f*
/Users/artyom/go/src/mod/cache/download/golang.org/x/sync/@v/v0.0.0-20180314180146-1d60e4601c6f.info
/Users/artyom/go/src/mod/cache/download/golang.org/x/sync/@v/v0.0.0-20180314180146-1d60e4601c6f.mod
```

**Comment by artyom** [comment]
Removal of `~/go/src/mod` directory helps.

**Comment by warthog618** [comment]
I'm seeing this as well.  Rolling vgo back one commit to 1d3e6141fcb3b8adaf02512d0c049a7aa5e386b4 fixes it for me, so it looks like the bbcfaa0c1b10a3e580b80b1d575ce55054b80a23 commit is the culprit.

**Comment by folago** [comment]
I hit it as well, as @artyom said removal of `~/go/src/mod` fixed it for me.
